src/mm.rs

```diff
@@ -73,20 +76,41 @@ pub fn load_user_app(app_name: &str) -> AxResult<(VirtAddr, VirtAddr, AddrSpace)
         // TDOO: flush the I-cache
     }
 
-    let ustack_top = uspace.end();
-    let ustack_vaddr = ustack_top - crate::USER_STACK_SIZE;
+    let ustack_base = uspace.end();
+    let ustack_size = crate::USER_STACK_SIZE;
+    let ustack_top = ustack_base - ustack_size;
     debug!(
         "Mapping user stack: {:#x?} -> {:#x?}",
-        ustack_vaddr, ustack_top
+        ustack_top, ustack_base
     );
+    // FIXME: Add more arguments and environment variables
+    let args = vec![app_name.to_string()];
```

Comment:
Try to make the type of the `args` parameter in `get_app_stack_region` a slice, to avoid heap allocation here.